File: lib/segmento.rb
# ═══════════════════════════════════════════════════════════════════
# SEGMENTO DE MERCADO — o "coringa" do sistema.
#
# A marca decide o segmento (config/brands/<marca>.yml → segmento:) e o
# ambiente pode forçar com SEGMENTO=... na VPS. Sem nada configurado vale
# a pré-seleção de fábrica: clinica (oftalmológica) — o comportamento de
# sempre, sem mudar um pixel da CEVICO.
#
# O segmento é um pacote EDITÁVEL em config/segmentos/<id>.yml:
# terminologia (paciente→cliente...), profissionais, unidades, janelas
# de agenda, modalidades, tabela de preços padrão, jornada, metas e os
# prompts dos robôs de IA. Criar um mercado novo = criar um yml novo,
# sem tocar em código.
#
# Regra de ouro: TODO valor lido daqui tem o preset da clínica como
# fallback natural — o yml da clínica guarda exatamente os valores que
# antes estavam chumbados no código.
# ═══════════════════════════════════════════════════════════════════
module Segmento
  FALLBACK_ID = 'clinica'.freeze

  module_function

  def config
    @config ||= begin
      wanted = ENV['SEGMENTO'].to_s.strip.downcase.presence ||
               (defined?(MARCA) ? MARCA['segmento'].to_s.strip.downcase.presence : nil) ||
               FALLBACK_ID
      pacote = load_yml(wanted)
      if pacote.nil? && wanted != FALLBACK_ID
        Rails.logger&.warn("[segmento] pacote '#{wanted}' não existe — usando #{FALLBACK_ID}")
        pacote = load_yml(FALLBACK_ID)
      end
      (pacote || {}).freeze
    end
  end

  def load_yml(seg_id)
    file = Rails.root.join('config', 'segmentos', "#{seg_id}.yml")
    return nil unless File.exist?(file)

    YAML.safe_load(File.read(file))
  end

  def id
    config['id'] || FALLBACK_ID
  end

  def clinica?
    id == 'clinica'
  end

  # ── Terminologia ────────────────────────────────────────────
  # termo(:cliente) → 'paciente' (clínica) / 'cliente' (empresa)
  def termo(chave)
    config.dig('termos', chave.to_s) || chave.to_s
  end

  # 'paciente' → 'Paciente' (só a primeira letra; preserva acentos)
  def termo_cap(chave)
    t = termo(chave)
    t.blank? ? t : t[0].upcase + t[1..]
  end

  # frases prontas (rótulos compostos onde gênero/concordância mudam)
  def frase(chave, padrao = nil)
    config.dig('frases', chave.to_s).presence || padrao || chave.to_s
  end

  # ── Profissionais (os "médicos" do segmento) ────────────────
  def profissionais
    Array(config['profissionais'])
  end

  # { 'Dr. Gustavo Bittar' => 'gustavo|bittar', ... } — nome oficial →
  # padrão de grafias tolerante (campo doctor das tasks é texto livre)
  def profissionais_grafias
    profissionais.each_with_object({}) do |p, map|
      next if p['nome'].blank?

      map[p['nome']] = p['grafias'].presence || Regexp.escape(p['nome'])
    end
  end

  # ── Unidades ────────────────────────────────────────────────
  def unidades
    Array(config['unidades'])
  end

  def unidade_keys
    unidades.filter_map { |u| u['key'] }
  end

  def unit_label(key)
    unidades.find { |u| u['key'] == key.to_s }&.dig('nome') || key.to_s
  end

  # { 'tatuape' => 'Tatuapé', ... } — substituto dos UNIT_LABELS chumbados
  def unit_labels
    unidades.each_with_object({}) { |u, map| map[u['key']] = u['nome'] if u['key'] }
  end

  # ── Agenda ──────────────────────────────────────────────────
  def janelas_padrao
    Array(config['janelas_padrao'])
  end

  def modalidades
    Array(config['modalidades'])
  end

  # ── Tabela de preços padrão do segmento ─────────────────────
  def precos_padrao
    Array(config['precos'])
  end

  # ── Jornada / metas / estrutura ─────────────────────────────
  def jornada
    config['jornada'] || {}
  end

  def pilares
    Array(config['pilares'])
  end

  def indicadores
    config['indicadores'] || {}
  end

  def indicadores_percentuais
    config['indicadores_percentuais'] || {}
  end

  def financeiro_categorias
    config['financeiro_categorias'] || {}
  end

  def estoque_categorias
    config['estoque_categorias'] || {}
  end

  # ── Robôs de IA ─────────────────────────────────────────────
  # prompt do segmento para um serviço ('conversation_insight',
  # 'instagram_agent'...). nil = o serviço usa o SYSTEM_PROMPT chumbado
  # (que É o preset da clínica — por isso o clinica.yml não traz prompts).
  def prompt(chave)
    config.dig('prompts', chave.to_s).presence
  end

  # trava de segurança do segmento; nil = usa a trava chumbada (clínica)
  def guardrail(respondedor:)
    config.dig('guardrails', respondedor ? 'respondedor' : 'operacional').presence
  end

  # contexto do negócio p/ {{CONTEXTO_DO_NEGOCIO}} nos prompts genéricos
  def contexto_negocio
    config['contexto'].to_s
  end

  # ── Fatia enviada ao navegador (window.SEGMENTO) ────────────
  # sem prompts/guardrails: o front não precisa e o HTML fica leve
  def frontend
    config.slice(
      'id', 'nome', 'termos', 'frases', 'profissionais', 'unidades',
      'janelas_padrao', 'modalidades', 'problemas', 'procedimentos',
      'metas', 'paineis'
    )
  end
end

File: app/javascript/dashboard/helper/segmento.js
// Pacote de SEGMENTO (sistema coringa) — injetado pelo Rails em
// window.SEGMENTO (vueapp.html.erb) a partir de config/segmentos/<id>.yml.
// A marca escolhe o segmento (cevico → clinica, life → empresa); sem
// pacote (testes, contexto de widget) vale o preset clínica — os mesmos
// textos que sempre estiveram chumbados no frontend.

const SEG = (typeof window !== 'undefined' && window.SEGMENTO) || {};

export const SEGMENTO = SEG;
export const segmentoId = SEG.id || 'clinica';
export const isClinica = segmentoId === 'clinica';

// termo('cliente') → 'paciente' na clínica / 'cliente' na empresa.
// Sem pacote, a chave e o termo da clínica coincidem com o texto antigo.
const TERMOS_CLINICA = {
  cliente: 'paciente',
  clientes: 'pacientes',
  profissional: 'médico',
  profissionais: 'médicos',
  atendimento: 'consulta',
  atendimentos: 'consultas',
  venda: 'cirurgia',
  vendas: 'cirurgias',
  empresa: 'clínica',
  unidade: 'unidade',
  unidades: 'unidades',
};

export const termo = chave => SEG.termos?.[chave] || TERMOS_CLINICA[chave] || chave;

// 'paciente' → 'Paciente' (só a primeira letra; preserva acentos)
export const termoCap = chave => {
  const t = termo(chave);
  return t ? t.charAt(0).toUpperCase() + t.slice(1) : t;
};

// frase('novo_atendimento', 'Nova consulta') → rótulo composto do
// segmento; o padrão passado é sempre o texto da clínica (o de sempre)
export const frase = (chave, padrao) => SEG.frases?.[chave] || padrao || chave;

// metas do segmento (ex.: meta('vendas_mes', 100))
export const meta = (chave, padrao) => {
  const v = Number(SEG.metas?.[chave]);
  return Number.isFinite(v) && v > 0 ? v : padrao;
};

// nome de quem cuida de um painel do Início ('' = sem nome exibido)
export const painelQuem = (chave, padrao = '') => {
  const paineis = SEG.paineis || {};
  return chave in paineis ? String(paineis[chave] || '') : padrao;
};

// listas do segmento com fallback (usadas por cevicoAgenda.js e telas)
export const listaSegmento = (chave, padrao = []) =>
  Array.isArray(SEG[chave]) && SEG[chave].length ? SEG[chave] : padrao;

// nome do sistema no hero do Início: clínica mantém o "CEVICO S.I" de
// sempre; outros segmentos usam o nome da instalação (pacote de marca)
export const nomeSistemaHero = () => {
  if (SEG.frases?.nome_sistema_hero) return SEG.frases.nome_sistema_hero;
  if (isClinica) return 'CEVICO S.I';
  return (
    (typeof window !== 'undefined' && window.globalConfig?.INSTALLATION_NAME) ||
    'seu sistema'
  );
};
